### ISSUE TITLE
Going Deeper with Convolutions

### ISSUE THREAD

**Issue by wrryu09** [GDSC-Ewha-4th/Study-paperReview#7]
### Going Deeper with Convolutions

**Abstract**
_    목적: ImageNet Large-Scale Visual Recognition Challenge 2014 (이하 ILSVRC14)에 이미지 classification, 
detection을 위한 deep convolutional neural network architecture 제안

**Introduction**
_    의도: computer vision을 위한 더 효율적인 deep neural network architecture의 개발

**3. Motivation and High Level Considerations**
    deep neural network quality를 증가시키는 가장 좋은 방법은 network 크기를 키우는 것인데, 두가지 주요 문제가 있음.
    1. overfitting에 취약하다 (training set의 수 제한 시 특히)
    2. computationcal resource use 크게 증가

**sol.** fully connected layers를 sparse ones로 바꾸고, sparsity를 도입하는 것.

하지만 현대의 컴퓨팅 인프라는 비정형 sparse data 구조의 산술 계산에 아주 비효율적. 
현재의 computer vision을 위한 구조들은 uniform structure를 사용한다. 

**filter-level sparsity를 신경쓰면서도... dense matrices의 computation을 활용해 현재 하드웨어를 사용하는 중간 단계는 없을까?
sparse matrices들을 clustering -> dense submatrices 하는 방법의 performance가 꽤 괜찮았다!**

**4. Architectural Details**
Main idea :  **optimal local sparse structure를 어떻게 readily available dence component로 cover** 할 수 있을까

 layer-by layer construction은 마지막 레이어의 상관 통계를 분석해서 그것들을 높은 상관관계를 가지는 유닛들의 그룹으로 cluster 하는 것이다. 이렇게 하면 다음 레이어의 유닛을 구성하는 cluster들은 전 레이어의 유닛들과 연결되어 있다. 
patch-alignment issue를 피하기 위해 필터 사이즈를 1x1, 3x3, 5x5로 제한했다. (필요보다는 편리를 위한 것)
suggested architecture는 다음 스테이지의 인풋을 구성하는 하나의 결과 벡터에 output filter bank들을 이어 만든.. 그 모든 레이어의 콤비네이션

higher abstraction을 가진 요소는 higher layers에 의해 capture되고, spatial concentration은 감소한다. 
따라서 3x3, 5x5 비율을 가진 convolutions는 higher layer로 갈수록 증가되어야 한다.

Inception architecture의 두 번째 아이디어
**다른 곳보다 computational requirements가 증가하는 곳의 차원을 줄이는 것.**  
representation은 대부분의 경우 sparse하게 유지되야 하고 결합되어야 할 때만 신호들을 compress해야 한다. 
그래서, 1x1 convolutions는 비싼 3x3과 5x5가 사용되기 전에 사용된다.

**6. Training Methodology**
GoogLeNet network는 DistBelidf distributed machine learning system으로  data-parallelism을 사용해 train되었다. CPU based implementation만을 사용했기 때문에, 적은 하이엔드 GPU를 사용해 GoogLeNet 네트워크가 집합되도록 해야 했고, 주 제약은 memory usage다. 

7. ILSVRC 2014 Classification Challenge Setup and Results
-더 좋은 결과를 얻기 위해 아래 기술들을 사용함

-  같은 GoogLeNet 모델의 7가지 버전을 독립적으로 트레이닝함. 모두 같은 초기화를 거쳤고, 다른 부분은 sampling methodologies, randomized input image order.

-   adopted more aggressive cropping

-  softmax probabilities are averaged over multiple crops and over all the individual classifiers to obtain the final prediction.

8. ILSVRC 2014 Detection Challenge Setup and Results
실패를 줄이기 위해 super pixel 사이즈를 2x로 증가시켰는데, 이는 선택적 탐색 알고리즘의 proposal를 반으로 줄인다. 각 region을 classify 할 때는 여섯 개의 GoogLeNet을 합친 것을 사용했다. 정확도를 조금 올릴 수 있었으며 (정확한 수치는 원문 참고) R-CNN과 달리 bounding box regression을 시간 부족으로 사용하지 않았음.

9. Conclusions
**expected optimal sparse structure을 readily available dense building blocks을 이용해 approximating 하는 것은 CV의 neural networks를 발전시키는 데에 효과가 있음.**

**Comment by yj-song131** [comment]
잘 읽었습니다!! GoogLeNet에 관심이 많았는데 저도 이 논문 읽어봐야겠어요 👀

**Comment by Soyeon-Cha** [comment]
리뷰 잘 읽었습니다 :)  아이디어의 배경과 아이디어를 잘 정리해주셔서 읽으면서 이 논문에 흥미가 생겼습니다. 다음에 이 논문 읽기에 도전해볼게요!!
